### PR TITLE
Np 466 update publication

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: modify-publication/src/test/java/no/unit/nva/publication/ModifyPublicationHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        environmentVariables.set(ALLOWED_ORIGIN_ENV, "*");
+        environmentVariables.set(API_HOST_ENV, "localhost:3000");
+        environmentVariables.set(API_SCHEME_ENV, "http");
+        ModifyPublicationHandler modifyPublicationHandler = new ModifyPublicationHandler();
+        assertNotNull(modifyPublicationHandler);
+    }
+
+    @Test
+    public void testOkResponse() throws IOException, InterruptedException {
+        ModifyResourceService modifyResourceService = mock(ModifyResourceService.class);
```
[the remark this answers is not in the document]
Comment (janbj) [reply modify-publication/src/test/java/no/unit/nva/publication/ModifyPublicationHandlerTest.java]: I moved them to class and annotated with Mock

File: modify-publication/src/test/java/no/unit/nva/publication/ModifyPublicationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        assertEquals(SC_OK, gatewayResponse.getStatusCode());
+        Assert.assertTrue(gatewayResponse.getHeaders().keySet().contains(CONTENT_TYPE));
+        Assert.assertTrue(gatewayResponse.getHeaders().keySet().contains(ACCESS_CONTROL_ALLOW_ORIGIN));
+    }
+
+    @Test
+    public void testIdentifiersInPathParametersAndBodyAreNotTheSame() throws IOException {
+        ModifyResourceService modifyResourceService = mock(ModifyResourceService.class);
```
[the remark this answers is not in the document]
Comment (janbj) [reply modify-publication/src/test/java/no/unit/nva/publication/ModifyPublicationHandlerTest.java]: Moved most of the repeated code to class level

File: modify-publication/src/main/java/no/unit/nva/publication/ModifyPublicationHandler.java [hunk 3]
```diff
@@ -117,18 +112,25 @@ public void handleRequest(InputStream input, OutputStream output, Context contex
...
+            writeErrorResponse(output, BAD_GATEWAY, e);
         } catch (Exception e) {
             logError(e);
-            objectMapper.writeValue(output, new GatewayResponse<>(objectMapper.writeValueAsString(
-                    Problem.valueOf(INTERNAL_SERVER_ERROR, e.getMessage())), headers(), SC_INTERNAL_SERVER_ERROR));
+            writeErrorResponse(output, INTERNAL_SERVER_ERROR, e);
         }
     }
 
+    private void writeErrorResponse(OutputStream output, Status status, String message) throws IOException {
```
Comment (axthosarouris) [inline modify-publication/src/main/java/no/unit/nva/publication/ModifyPublicationHandler.java]: I was thinking of a library, but we can discuss it later :)

Comment (janbj) [reply modify-publication/src/main/java/no/unit/nva/publication/ModifyPublicationHandler.java]: Yes, it was first step of creating a common super class for publication handlers

**Pull request merged**